File: lines-in-plane.swift
func getRegionsInPlaneByLinesRecurrently(numLines: UInt) -> UInt{
  if numLines == UInt.min { return 1 }
  return getRegionsInPlaneByLinesRecurrently(numLines-1) + numLines
} //O(n)

func getRegionsInPlaneByLinesClosed(numLines: UInt) -> UInt {
  return (numLines*(numLines+1)) / 2 + 1
} //O(1)

File: lines-in-plane.js
function getRegionsInPlaneByLinesRecurrently(numLines){
  if (numLines == 0) return 1;
  return getRegionsInPlaneByLinesRecurrently(numLines-1) + numLines;
} // O(n)

function getRegionsInPlaneByLinesClosed(numLines){
  return (numLines * (numLines+1))/ 2 + 1;
} // O(1)
